sc-memory/tests/sc-memory/templates/sc-templates-common.cpp

```diff
@@ -423,3 +424,90 @@ TEST_F(ScTemplateCommonTest, big_template_smoke)
   for (size_t i = 0; i < searchResult.Size(); ++i)
     EXPECT_TRUE(elements.find(searchResult[i]["_el"]) != elements.end());
 }
+
+TEST_F(ScTemplateCommonTest, CycledTemplateSmoke)
+{
+  ScAddr const & classAddr = m_ctx->CreateNode(ScType::NodeConstClass);
+  ScAddr const & sourceNodeAddr = m_ctx->CreateNode(ScType::NodeConst);
+  m_ctx->CreateEdge(ScType::EdgeAccessConstPosPerm, classAddr, sourceNodeAddr);
+
+  size_t const constrCount = 5;
+  size_t const tripleCount = 20;
+  for (size_t i = 0; i < constrCount; ++i)
+  {
+    ScAddr const & targetNodeAddr = m_ctx->CreateNode(ScType::NodeConst);
+
+    ScTemplate genTempl;
+    genTempl.Triple(
+        sourceNodeAddr,
+        ScType::EdgeAccessVarPosPerm,
+        targetNodeAddr
+    );
+
+    for (size_t j = 0; j < tripleCount; ++j)
+    {
+      ScTemplateGenResult result;
+      EXPECT_TRUE(m_ctx->HelperGenTemplate(genTempl, result));
+    }
+  }
+
+  ScTemplate searchTempl;
+  searchTempl.Triple(
+      classAddr,
+      ScType::EdgeAccessVarPosPerm,
+      ScType::NodeVar >> "_source"
+  );
+  searchTempl.Triple(
+      "_source",
+      ScType::EdgeAccessVarPosPerm,
+      ScType::NodeVar >> "_target"
+  );
+  for (size_t i = 0; i < tripleCount - 1; ++i)
```

Comment:
in witch cases we can find twentieth edge?

Reply:
Please, clarify your question